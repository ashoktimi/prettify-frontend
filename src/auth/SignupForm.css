.SignupForm {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  max-width: 50%;
  padding: 20px;
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.2);
}

.SignupForm h2 {
  margin-bottom: 20px;
  font-size: 24px;
  text-align: center;
  color: #007bff;
}

.SignupForm label {
  display: block;
  margin-bottom: 10px;
  font-weight: bold;
}

.SignupForm input {
  width: 100%;
  padding: 8px 10px;
  margin-bottom: 20px;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-size: 16px;
}

.SignupForm button {
  background-color: #007bff;
  color: #fff;
  padding: 10px 20px;
  border: none;
  border-radius: 4px;
  font-size: 16px;
  cursor: pointer;
}

.SignupForm button:hover {
  background-color: #0069d9;
}

.SignupForm .alert {
  margin-bottom: 20px;
  border-radius: 4px;
  padding: 10px;
  color: #721c24;
  background-color: #f8d7da;
  border: 1px solid #f5c6cb;
}
