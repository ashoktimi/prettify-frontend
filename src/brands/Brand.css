.Head-Name{
 text-transform: uppercase;
 color: white;
 background-color: rgb(37, 150, 190);
 padding: 5px;
 border-radius: 5px;
 letter-spacing: 5px;
 font-size: 25px;
 padding-left: 10px;
}
