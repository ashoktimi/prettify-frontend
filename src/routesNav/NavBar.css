
.NavBar-link {
    color: #333333;
    text-decoration:none;
    margin-left: 1.5rem;
    font-weight: bold;
    font-family: 'Heebo', sans-serif;
    margin-top: 15px;
    font-size: 20px;
} 

.NavBar-dropdown{
    font-weight: bold;
    font-family: 'Heebo', sans-serif;
    margin-left: 1.5rem;
}

.NavBar-dropdown > .dropdown-toggle::after {
  display: none;
}

.nav-link{
    color: #333333;
    padding-top: 5px;
    margin-top: 0;
    font-size: 20px;
}

.nav-item{
  margin-top: 10px;
}
.NavBar-dropdown-link{
    color: #333333;
    font-family: 'Heebo', sans-serif;
    text-decoration: none;
    text-transform: capitalize;
}

.Navbar-bag{
    color:black;
    text-decoration: none;
    margin-right: 15px;
    margin-left: 2.5rem;
    margin-top: 10px;
}

.Navbar-bag:hover{
  color: black;
}

.bag-quantity {
  border-radius: 50%;
  font-size: 20px;
  font-weight: 700;
}
.Profile-dropdown{
  padding-top: 5px;
}
.Profile-dropdown > .dropdown-toggle::after{
  display: none;
}

.Navbar-logout{
  color: #333333;
  text-decoration: none;
  font-weight: bold;
  font-size: 20px;
  font-family: 'Heebo', sans-serif;
  margin-top: 15px;
  margin-left: 15px;
}

.Navbar-logout:hover{
  color: black;
}
