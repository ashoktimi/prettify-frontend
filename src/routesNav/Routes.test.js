import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { UserProvider,  commonBeforeEach, commonAfterEach} from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import { BrowserRouter } from "react-router-dom";
import cartReducer from '../cart/cartSlice';
import Routers from "./Routers";
import NavBar from './NavBar';
import Home from "../home/Home";
import Login from '../auth/LoginForm';
import SignupForm from "../auth/SignupForm";
import Brand from "../brands/Brand";
import BrandList from "../brands/BrandList";
import ProductList from "../products/ProductList";
import Product from "../products/Product";
import ProductDetail from "../products/ProductDetail";
import CategoryList from "../categories/CategoyList";
import Category from "../categories/Category";
import TagLists from "../tags/TagList";
import Tag from "../tags/Tag";
import TypeList from "../types/TypeList";
import Type from "../types/Type";
import NotFound from "./NotFound";
import CheckoutPage from "../cart/Checkout";


beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('Routes', () => {
const store = configureStore({
  reducer: {
    cart: cartReducer
  }
});

test("renders without crashing", function () {
  render(
    <Provider store={store}>
      <Routers />
    </Provider>
  );
});

test("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <Routers />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('NavBar component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <UserProvider>
          <NavBar />
        </UserProvider>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('Home component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <Home />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('Brand component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <Brand />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('BrandList component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <BrandList />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('ProductList component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <ProductList />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('Product component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <UserProvider>
          <Product />
        </UserProvider>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('CategoryList component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
        <UserProvider>
          <CategoryList />
        </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});


test('Category component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <Category />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('ProductDetail component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <UserProvider>
          <ProductDetail />
        </UserProvider>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('Type component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <Type />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});



test('TypeList component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <TypeList />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});


test('Tag component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <Tag />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('TagLists component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <TagLists />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('NotFound component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <UserProvider>
        <NotFound />
      </UserProvider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('CheckoutPage component renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <UserProvider>
          <CheckoutPage />
        </UserProvider>
      </Provider>
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test('Routes to login page', () => {
  const { getByText } = render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>
  );
  expect(getByText("Log In")).toBeInTheDocument();
});
test('Routes to signup page', () => {
  const { getByText } = render(
    <BrowserRouter>
      <SignupForm />
    </BrowserRouter>
  );
  expect(getByText("Sign Up")).toBeInTheDocument();
});
});