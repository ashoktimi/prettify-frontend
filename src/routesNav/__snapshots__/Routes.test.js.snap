// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Brand component renders correctly 1`] = `
<div>
  <nav
    class="NavBtm navbar navbar-expand navbar-light"
  />
</div>
`;

exports[`Routes BrandList component renders correctly 1`] = `<div />`;

exports[`Routes Category component renders correctly 1`] = `
<div>
  <nav
    class="NavBtm navbar navbar-expand navbar-light"
  />
</div>
`;

exports[`Routes CategoryList component renders correctly 1`] = `<div />`;

exports[`Routes CheckoutPage component renders correctly 1`] = `
<div>
  <div
    style="width: 70%; margin-left: 15%; margin-top: 5rem; display: flex; justify-content: space-between;"
  >
    <form
      class=""
    >
      <h2>
        Personal Information
      </h2>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formFirstName"
            >
              First Name:
            </label>
            <input
              class="form-control"
              id="formFirstName"
              name="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formLastName"
            >
              Last Name:
            </label>
            <input
              class="form-control"
              id="formLastName"
              name="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <label
          class="form-label"
          for="formEmail"
        >
          Email:
        </label>
        <input
          class="form-control"
          id="formEmail"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formCountry"
            >
              Country:
            </label>
            <input
              class="form-control"
              id="formCountry"
              name="country"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formPostalCode"
            >
              Postal Code:
            </label>
            <input
              class="form-control"
              id="formPostalCode"
              name="postalCode"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <label
          class="form-label"
          for="formPhone"
        >
          Phone Number:
        </label>
        <input
          class="form-control"
          id="formPhone"
          name="phone"
          type="tel"
          value=""
        />
      </div>
      <h2>
        Payment Details
      </h2>
      <div>
        <label
          class="form-label"
          for="formCardNumber"
        >
          Card Number:
        </label>
        <input
          class="form-control"
          id="formCardNumber"
          name="cardNumber"
          type="text"
          value=""
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formExpMonth"
            >
              Expiration Month:
            </label>
            <input
              class="form-control"
              id="formExpMonth"
              name="expMonth"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formExpYear"
            >
              Expiration Year:
            </label>
            <input
              class="form-control"
              id="formExpYear"
              name="expYear"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formCVV"
            >
              CVV:
            </label>
            <input
              class="form-control"
              id="formCVV"
              name="cvv"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="billingAddress"
            >
              Billing Address:
            </label>
            <input
              class="form-control"
              id="billingAddress"
              name="billing"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="btn btn-warning"
        style="margin-top: 10px;"
        type="button"
      >
        Submit
      </button>
    </form>
    <div
      style="background-color: white; padding: 0px 3rem 5rem 3rem; width: 40%; margin-left: 1rem; margin-top: 5rem;"
    >
      <h5
        style="text-align: center; margin-bottom: 1.5rem; margin-top: 2rem;"
      >
        IN YOUR BAG
      </h5>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <p>
          Subtotal:
        </p>
        <p>
          $
          0
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <p>
          Estimated Shipping:
        </p>
        <p>
          0
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <p>
          Estimated Tax:
        </p>
        <p>
          0
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <h5>
          Total:
        </h5>
        <h5>
           $
          0
        </h5>
      </div>
    </div>
  </div>
</div>
`;

exports[`Routes Home component renders correctly 1`] = `<div />`;

exports[`Routes NavBar component renders correctly 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-light sticky-top"
    style="background-color: rgb(193, 111, 184); padding: 1rem; height: 4rem;"
  />
  <nav
    class="navbar navbar-expand-bg navbar-light"
    style="padding: 0px;"
  >
    <div
      class="container-fluid"
    >
      <button
        aria-controls="offcanvasNavbar-expand"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="NavBar-name-div"
      >
        <a
          class="NavBar-name"
          href="/products"
        >
          Prettify
        </a>
      </div>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand"
        class="offcanvas offcanvas-start"
        id="offcanvasNavbar-expand"
      >
        <div
          class="offcanvas-header"
        >
          <div
            class="offcanvas-title h5"
            id="offcanvasNavbarLabel-expand"
            style="color: purple; left: 0px; right: 0px; margin: auto;"
          >
            Prettify
          </div>
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="offcanvas-body"
        >
          <ul
            class="Toggler-ul"
          >
            <li>
              <a
                href="/products"
              >
                ALL
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                 BRANDS 
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                CATEGORY
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                TAGS
              </a>
            </li>
          </ul>
        </div>
        <a
          class="Toggler-bag"
          href="/"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
            />
          </svg>
        </a>
        <a
          class="Toggler-logout"
          href="/"
          style="margin-bottom: 35px;"
        >
          Log out
        </a>
      </div>
    </div>
  </nav>
  <nav
    class="navbar navbar-expand-lg navbar-light sticky-top"
  >
    <div
      class="container"
    >
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="mr-auto navbar-nav"
          style="left: 0px; right: 0px; margin: auto;"
        >
          <a
            class="NavBar-link"
            href="/products"
          >
            ALL
          </a>
          <a
            class="NavBar-link"
            href="/types/nail/nail_polish"
          >
            NAIL
          </a>
          <div
            class="NavBar-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              EYES
            </a>
          </div>
          <div
            class="NavBar-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              LIPS
            </a>
          </div>
          <div
            class="NavBar-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              FACE
            </a>
          </div>
          <div
            class="nav-item"
            style="padding-top: 5px;"
          >
            <a
              class="NavBar-link"
              href="/"
            >
              BRANDS
            </a>
          </div>
          <div
            class="nav-item"
            style="padding-top: 5px;"
          >
            <a
              class="NavBar-link"
              href="/"
            >
              CATEGORIES
            </a>
          </div>
          <div
            class="nav-item"
            style="padding-top: 5px;"
          >
            <a
              class="NavBar-link"
              href="/"
            >
              TAGS
            </a>
          </div>
          <a
            class="Navbar-bag"
            href="/"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
              />
            </svg>
            <span
              class="bag-quantity"
            >
              0
            </span>
          </a>
          <div
            class="Profile-dropdown nav-item dropdown"
            style="margin-top: 10px;"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
          </div>
          <a
            class="Navbar-logout"
            href="/"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Routes NotFound component renders correctly 1`] = `
<div>
  <div>
    <h1>
      404 Not Found
    </h1>
    <p>
      The page you are looking for could not be found.
    </p>
  </div>
</div>
`;

exports[`Routes Product component renders correctly 1`] = `
<div>
  <div>
    <div
      class="Product-div"
    >
      <div
        class="container"
        style="margin-top: 5rem;"
      >
        <div
          class="justify-content-center align-items-center row"
        >
          <div
            class="text-center mb-4 mb-md-0 col-md-6 col-12"
          >
            <img
              class="ProductDetail-image"
              src="spinning image.png"
              style="width: 30rem; height: 30rem; background-color: white;"
            />
          </div>
          <div
            class="col-md-6 col-12"
          >
            <div
              class="ProductDetail-info"
            >
              <h5
                class="ProductDetail-name"
              />
              <p
                class="ProductDetail-description"
              />
              <small
                class="ProductCard-price"
              >
                 
                NaN
                 
              </small>
              <small
                class="ms-2 text-muted priceBefore"
              >
                 before 
                NaN
                 
              </small>
              <div
                class="ProductDetail-rating"
              >
                <small>
                  (
                  NaN
                  )
                </small>
                <p>
                  (
                  ) Reviews
                </p>
              </div>
              <button
                class="btn btn-warning"
                style="margin-top: 2rem;"
                type="button"
              >
                ADD To CART
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Routes ProductDetail component renders correctly 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 5rem;"
  >
    <div
      class="justify-content-center align-items-center row"
    >
      <div
        class="text-center mb-4 mb-md-0 col-md-6 col-12"
      >
        <img
          class="ProductDetail-image"
          src="spinning image.png"
          style="width: 30rem; height: 30rem; background-color: white;"
        />
      </div>
      <div
        class="col-md-6 col-12"
      >
        <div
          class="ProductDetail-info"
        >
          <h5
            class="ProductDetail-name"
          />
          <p
            class="ProductDetail-description"
          />
          <small
            class="ProductCard-price"
          >
             
             
          </small>
          <small
            class="ms-2 text-muted priceBefore"
          >
             before 
             
          </small>
          <div
            class="ProductDetail-rating"
          >
            <small>
              (
              )
            </small>
            <p>
              (
              ) Reviews
            </p>
          </div>
          <button
            class="btn btn-warning"
            style="margin-top: 2rem;"
            type="button"
          >
            ADD To CART
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Routes ProductList component renders correctly 1`] = `
<div>
  <div
    class="LoadingSpinner"
  >
    Loading ...
  </div>
</div>
`;

exports[`Routes Tag component renders correctly 1`] = `
<div>
  <nav
    class="NavBtm navbar navbar-expand navbar-light"
  />
</div>
`;

exports[`Routes TagLists component renders correctly 1`] = `<div />`;

exports[`Routes Type component renders correctly 1`] = `
<div>
  <nav
    class="NavBtm navbar navbar-expand navbar-light"
  />
</div>
`;

exports[`Routes TypeList component renders correctly 1`] = `
<div>
  <div
    class="LoadingSpinner"
  >
    Loading ...
  </div>
</div>
`;

exports[`Routes matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Toastify"
    />
    <nav
      class="navbar navbar-expand navbar-light sticky-top"
      style="background-color: rgb(193, 111, 184); padding: 1rem; height: 4rem;"
    />
    <nav
      class="navbar navbar-expand-bg navbar-light"
      style="padding: 0px;"
    >
      <div
        class="container-fluid"
      >
        <button
          aria-controls="offcanvasNavbar-expand"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="NavBar-name-div"
        >
          <a
            class="NavBar-name"
            href="/"
            style="margin-left: 10rem;"
          >
            Prettify
          </a>
        </div>
        <div
          style="margin-right: 5rem;"
        >
          <a
            class="NavBar-auth_Btn"
            href="/login"
          >
            Login
          </a>
          <a
            class="NavBar-auth_Btn"
            href="/signup"
          >
            Signup
          </a>
        </div>
        <div
          aria-labelledby="offcanvasNavbarLabel-expand"
          class="offcanvas offcanvas-start"
          id="offcanvasNavbar-expand"
        >
          <h3
            class="Toggler-h3"
          >
            Please login to see.
          </h3>
        </div>
      </div>
    </nav>
    <div
      style="margin-top: 5rem; width: 80%; margin-left: 10%;"
    >
      <div
        class="Home"
      >
        <h2>
          Welcome!!
        </h2>
        <p
          class="lead"
        >
          All brands makeup items in one, convenient place. Get started by signing in or creating an account.
        </p>
        <p>
          Explore our collection of makeup items and find your perfect match.
        </p>
        <h5>
          Start Shopping Today
        </h5>
        <a
          class="btn btn-primary"
          href="/login"
        >
          Shop Now
        </a>
      </div>
      <div
        class="feature-boxes"
      >
        <div
          class="feature-box"
        >
          <h5
            style="text-align: center;"
          >
            Explore Our Features
          </h5>
          <ul>
            <li
              class="lead"
            >
              Search and filter different makeup products.
            </li>
            <li
              class="lead"
            >
              Create a personalized makeup bag with your favorite items.
            </li>
            <li
              class="lead"
            >
              Get recommendations for products based on your skin type and preferences.
            </li>
          </ul>
        </div>
        <div
          class="feature-box"
        >
          <h5
            style="text-align: center;"
          >
            Why Choose Our App?
          </h5>
          <ul>
            <li
              class="lead"
            >
              We offer a wide selection of makeup items from different brands.
            </li>
            <li
              class="lead"
            >
              Our platform is easy to use and user-friendly.
            </li>
            <li
              class="lead"
            >
              We prioritize the safety and quality of our products.
            </li>
          </ul>
        </div>
        <div
          class="feature-box"
        >
          <h5
            style="text-align: center;"
          >
            Join Our Community
          </h5>
          <p>
            Connect with other makeup enthusiasts and share your beauty journey with us on social media.
          </p>
          <div
            style="display: flex; justify-content: center;"
          >
            <button
              class="Home-social"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
                />
              </svg>
            </button>
            <button
              class="Home-social"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.026 15c6.038 0 9.341-5.003 9.341-9.334 0-.14 0-.282-.006-.422A6.685 6.685 0 0 0 16 3.542a6.658 6.658 0 0 1-1.889.518 3.301 3.301 0 0 0 1.447-1.817 6.533 6.533 0 0 1-2.087.793A3.286 3.286 0 0 0 7.875 6.03a9.325 9.325 0 0 1-6.767-3.429 3.289 3.289 0 0 0 1.018 4.382A3.323 3.323 0 0 1 .64 6.575v.045a3.288 3.288 0 0 0 2.632 3.218 3.203 3.203 0 0 1-.865.115 3.23 3.23 0 0 1-.614-.057 3.283 3.283 0 0 0 3.067 2.277A6.588 6.588 0 0 1 .78 13.58a6.32 6.32 0 0 1-.78-.045A9.344 9.344 0 0 0 5.026 15z"
                />
              </svg>
            </button>
            <button
              class="Home-social"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0C5.829 0 5.556.01 4.703.048 3.85.088 3.269.222 2.76.42a3.917 3.917 0 0 0-1.417.923A3.927 3.927 0 0 0 .42 2.76C.222 3.268.087 3.85.048 4.7.01 5.555 0 5.827 0 8.001c0 2.172.01 2.444.048 3.297.04.852.174 1.433.372 1.942.205.526.478.972.923 1.417.444.445.89.719 1.416.923.51.198 1.09.333 1.942.372C5.555 15.99 5.827 16 8 16s2.444-.01 3.298-.048c.851-.04 1.434-.174 1.943-.372a3.916 3.916 0 0 0 1.416-.923c.445-.445.718-.891.923-1.417.197-.509.332-1.09.372-1.942C15.99 10.445 16 10.173 16 8s-.01-2.445-.048-3.299c-.04-.851-.175-1.433-.372-1.941a3.926 3.926 0 0 0-.923-1.417A3.911 3.911 0 0 0 13.24.42c-.51-.198-1.092-.333-1.943-.372C10.443.01 10.172 0 7.998 0h.003zm-.717 1.442h.718c2.136 0 2.389.007 3.232.046.78.035 1.204.166 1.486.275.373.145.64.319.92.599.28.28.453.546.598.92.11.281.24.705.275 1.485.039.843.047 1.096.047 3.231s-.008 2.389-.047 3.232c-.035.78-.166 1.203-.275 1.485a2.47 2.47 0 0 1-.599.919c-.28.28-.546.453-.92.598-.28.11-.704.24-1.485.276-.843.038-1.096.047-3.232.047s-2.39-.009-3.233-.047c-.78-.036-1.203-.166-1.485-.276a2.478 2.478 0 0 1-.92-.598 2.48 2.48 0 0 1-.6-.92c-.109-.281-.24-.705-.275-1.485-.038-.843-.046-1.096-.046-3.233 0-2.136.008-2.388.046-3.231.036-.78.166-1.204.276-1.486.145-.373.319-.64.599-.92.28-.28.546-.453.92-.598.282-.11.705-.24 1.485-.276.738-.034 1.024-.044 2.515-.045v.002zm4.988 1.328a.96.96 0 1 0 0 1.92.96.96 0 0 0 0-1.92zm-4.27 1.122a4.109 4.109 0 1 0 0 8.217 4.109 4.109 0 0 0 0-8.217zm0 1.441a2.667 2.667 0 1 1 0 5.334 2.667 2.667 0 0 1 0-5.334z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
